spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/controller/StreamController.java

```diff
@@ -211,19 +213,28 @@ private void undeployStream(StreamDefinition stream) {
 		}
 	}
 
-
 	/**
-	 * Extension of {@link StreamDefinitionResource.Assembler} that
-	 * assembles {@link StreamDefinitionResource}s with stream status.
+	 * {@link org.springframework.hateoas.ResourceAssembler} implementation
+	 * that converts {@link StreamDefinition}s to {@link StreamDefinitionResource}s.
 	 */
-	class Assembler extends StreamDefinitionResource.Assembler {
+	public static class Assembler
+			extends ResourceAssemblerSupport<StreamDefinition, StreamDefinitionResource> {
+
+		public Assembler() {
+			super(StreamController.class, StreamDefinitionResource.class);
+		}
+
 		@Override
 		public StreamDefinitionResource toResource(StreamDefinition entity) {
+			return createResourceWithId(entity.getName(), entity);
+		}
 
-			StreamDefinitionResource resource = super.toResource(entity);
-			// todo: set stream status based on SPI status
```

Comment:
No, actually this PR needs to be rebased to pull the changes that addresses this comment. I am rebasing it now.